src/orchestrator/streaming.rs

```diff
@@ -352,9 +355,11 @@ async fn generation_broadcast_task(
 /// Consumes chunk broadcast stream, sends unary requests to a detector service,
 /// and sends chunk + responses to detection stream.
 #[instrument(skip_all)]
+#[allow(clippy::too_many_arguments)]
```

Comment:
This doesn't have to be done now, but isn't this a sign that we should rethink our structs?

Reply:
I created #237, so we don't forget about this.